test/system/irohad_test.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <rapidjson/document.h>
+#include <boost/filesystem.hpp>
+#include <boost/optional.hpp>
+#include <boost/process.hpp>
+#include <pqxx/pqxx>
+
+#include "common/files.hpp"
+#include "common/types.hpp"
+#include "main/iroha_conf_loader.hpp"
+
+using namespace boost::process;
+using namespace std::chrono_literals;
+using iroha::operator|;
+
+class IrohadTest : public testing::Test {
+ public:
+  void SetUp() override {
+    timeout = 1s;
+    setPaths();
+    auto config = parse_iroha_config(path_config_.string());
+    blockstore_path_ = config[config_members::BlockStorePath].GetString();
+    pgopts_ = config[config_members::PgOpt].GetString();
+  }
+  void TearDown() override {
+    iroha::remove_all(blockstore_path_);
+    dropPostgres();
+  }
+
+  std::string params(const boost::optional<std::string> &config_path,
+                     const boost::optional<std::string> &genesis_block,
+                     const boost::optional<std::string> &keypair_path) {
+    std::string res;
+    config_path | [&res](auto &&s) { res += " --config " + s; };
+    genesis_block | [&res](auto &&s) { res += " --genesis_block " + s; };
+    keypair_path | [&res](auto &&s) { res += " --keypair_name " + s; };
+    return res;
+  }
+
+  std::string setDefaultParams() {
+    return params(
+        path_config_.string(), path_genesis_.string(), path_keypair_.string());
+  }
+
+ private:
+  void setPaths() {
+    path_irohad_ = boost::filesystem::path(PATHIROHAD);
+    irohad_executable = path_irohad_ / "irohad";
+    path_example_ = path_irohad_.parent_path().parent_path() / "example";
+    path_config_ = path_example_ / "config.sample";
```

Comment:
test will not work locally with such config.

it contains "pg host=localhost", however when you run ./scripts/run-iroha-dev.sh postgres host is "postgres"